libs/cases/frontend/view/src/lib/components/detail-case/detail-case.component.ts

```diff
@@ -27,5 +30,135 @@ export class DetailCaseComponent {
         return _case.find((_case) => String(_case.id) === this.route.snapshot.params['id']);
     });
 
-    constructor(private route: ActivatedRoute) {}
+    isEditing = false;
+    wasValidated = false;
+    datePattern = /^\d{4}-\d{2}-\d{2}T00:00:00\.000Z$/;
+
+
+    constructor(private route: ActivatedRoute, private readonly _modalService: ModalService, private readonly toastService: ToastService) {
+    }
+
+    deleteCase() {
+        const caseId = this.mapCaseId(this.casedetail());
+        if (caseId !== undefined) {
+            this._browseFacade.deleteCase(caseId).subscribe();
+        }
+    }
+
+    mapCaseId(_case: ICaseResponse | undefined) {
+        if (_case === undefined) {
+            return undefined;
+        }
+        return {
+            id: _case.id,
+        };
+    }
+
+    async deleting() {
+        const instance = await this._modalService.open({
+            content: DeleteModalComponent,
+        });
+
+        // modal closes on confirm and dismisses on cancel
+        instance.onClose.on(() => {
+            this.deleteCase();
+        });
+    }
+
+    enableEditing() {
+        this.isEditing = true;
+    }
+
+    cancelEdit() {
+        this.isEditing = false;
+    }
```

Comment:
right now if you cancel the edit but already changed some values it will still show the edited (maybe wrong) values.
A cheesy way to fix this would be to just reload the page `window.location.reload();` but maybe you can find a better way to get the old values back

Reply:
keen eye. i implemented it and thanks for spotting that bug